### PR TITLE
feat(BA-3436): Implement Blue-Green deployment strategy

### PR DESCRIPTION
**Pull request by jopemachine**
Resolves BA-3436

## Summary

- Implement `BlueGreenStrategy.evaluate_cycle()` FSM with 8-step flow: classify routes → create green (INACTIVE) → wait provisioning → rollback if all failed → wait healthy → manual promote gate → promote delay check → atomic promotion (green→ACTIVE, blue→TERMINATING)
- Add `promote_route_ids` to `RouteChanges` for blue-green traffic switch, alongside existing `rollout_specs` (create) and `drain_route_ids` (terminate)
- Wire promote handling through the full stack: evaluator recording, applier `BatchUpdater` (traffic_status→ACTIVE), repository, and db_source transaction
- Add comprehensive unit tests (14 cases) covering all FSM branches: no green routes, provisioning wait, all-failed rollback, partial healthy, manual/auto/delayed promote, edge cases
- Uses `created_at` as conservative proxy for promote delay timing until `status_updated_at` is added to `RouteInfo`

## Key Design Decisions

- **INACTIVE green routes**: New-revision routes are created with `traffic_status=INACTIVE` and `traffic_ratio=0.0` so they don't receive traffic until explicitly promoted
- **Atomic promotion**: On COMPLETED, green routes are promoted (→ACTIVE) and blue routes are drained (→TERMINATING) in a single DB transaction
- **Promote ordering**: In `apply_strategy_mutations`, promote executes before drain to avoid a window where no routes serve traffic

## Test Plan

- [x] `test_blue_green.py` — 14 unit tests covering all 8 FSM steps
- [x] `test_applier.py` — existing applier tests pass with `promote` parameter addition
- [x] All deployment coordinator tests pass (8/8)
- [x] mypy type check passes
- [x] ruff lint/format passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[BA-3436]: https://lablup.atlassian.net/browse/BA-3436

----
📚 Documentation preview 📚: https://sorna--10050.org.readthedocs.build/en/10050/

----
📚 Documentation preview 📚: https://sorna-ko--10050.org.readthedocs.build/ko/10050/

### CODE REVIEW

File: src/ai/backend/manager/data/deployment/types.py [hunk 1]
```diff
@@ -583,6 +583,16 @@ def termination_priority(self) -> int:
             return 0
         return _HEALTH_TERMINATION_PRIORITY.get(self.health_status, 0)
 
+    @property
+    def is_ready(self) -> bool:
+        """Return True if the route is RUNNING and its health check is HEALTHY.
+
+        A ready route is eligible to receive traffic (subject to traffic_status).
```
Comment (jopemachine) [inline src/ai/backend/manager/data/deployment/types.py]: Would it be okay not to check the route traffic status?

File: src/ai/backend/manager/sokovan/deployment/strategy/blue_green.py [hunk 2]
```diff
@@ -28,5 +71,202 @@ def evaluate_cycle(
...
+
+    Ideally this would use ``status_updated_at`` to measure how long
+    the route has been HEALTHY, but that field is not yet available
+    on RouteInfo. ``created_at`` serves as a conservative proxy.
+    """
+    timestamps = [route.created_at for route in routes]
+    return max(timestamps) if timestamps else None
+
+
+def _build_route_creators(
```
Comment (jopemachine) [inline src/ai/backend/manager/sokovan/deployment/strategy/blue_green.py]: _build_green_routes_creators

File: src/ai/backend/manager/data/deployment/types.py (hunk 1 above)
Comment (jopemachine) [reply src/ai/backend/manager/data/deployment/types.py]: Intentionally excluded — blue-green green routes are INACTIVE until promotion, so checking `traffic_status == ACTIVE` would prevent them from ever being classified as ready. Updated the docstring to explain this in cc7a5ca0d.

File: src/ai/backend/manager/sokovan/deployment/strategy/blue_green.py (hunk 2 above)
Comment (jopemachine) [reply src/ai/backend/manager/sokovan/deployment/strategy/blue_green.py]: Renamed to `_build_green_route_creators` in cc7a5ca0d.
